marimo/_runtime/app/kernel_runner.py

```diff
@@ -113,6 +114,13 @@ def cache_output(
     def outputs(self) -> dict[CellId_t, Any]:
         return self._outputs
 
+    def are_outputs_cached(self, defs: dict[str, Any] | None) -> bool:
+        # The equality check is brittle but hashing isn't great either ...
+        return (defs == self._previously_seen_defs) and len(self.outputs) > 0
```

Comment:
Very brittle but distinguishes between `no` args and some args. Esp with a wrapper like mowidget, this should be fine